### ISSUE TITLE
error while installing lolcommits: :in `superclass': stack level too deep (SystemStackError)

### ISSUE THREAD

**Issue by monperrus** [lolcommits/lolcommits#431]
```
± gem install --user lolcommits
Fetching mini_magick-4.12.0.gem
Fetching methadone-1.9.5.gem
Fetching mercurial-ruby-0.7.12.gem
Fetching lolcommits-loltext-0.4.0.gem
Fetching open4-1.3.4.gem
Fetching launchy-2.5.2.gem
Fetching rchardet-1.8.0.gem
Fetching git-1.18.0.gem
Fetching lolcommits-0.16.4.gem
Successfully installed open4-1.3.4
Successfully installed mini_magick-4.12.0
Successfully installed methadone-1.9.5
Successfully installed mercurial-ruby-0.7.12
Successfully installed lolcommits-loltext-0.4.0
Successfully installed launchy-2.5.2
Successfully installed rchardet-1.8.0
Successfully installed git-1.18.0
Successfully installed lolcommits-0.16.4
Parsing documentation for open4-1.3.4
Installing ri documentation for open4-1.3.4
Parsing documentation for mini_magick-4.12.0
Installing ri documentation for mini_magick-4.12.0
```
```
Parsing documentation for methadone-1.9.5
Installing ri documentation for methadone-1.9.5
/usr/lib/ruby/3.0.0/rdoc/class_module.rb:702:in `superclass': stack level too deep (SystemStackError)
	from /usr/lib/ruby/3.0.0/rdoc/normal_class.rb:13:in `ancestors'
	from /usr/lib/ruby/3.0.0/rdoc/normal_class.rb:18:in `ancestors'
	from /usr/lib/ruby/3.0.0/rdoc/normal_class.rb:18:in `ancestors'
	from /usr/lib/ruby/3.0.0/rdoc/normal_class.rb:18:in `ancestors'
	from /usr/lib/ruby/3.0.0/rdoc/normal_class.rb:18:in `ancestors'
	from /usr/lib/ruby/3.0.0/rdoc/normal_class.rb:18:in `ancestors'
	from /usr/lib/ruby/3.0.0/rdoc/normal_class.rb:18:in `ancestors'
	from /usr/lib/ruby/3.0.0/rdoc/normal_class.rb:18:in `ancestors'
	 ... 10066 levels...
	from /usr/lib/ruby/vendor_ruby/rubygems/command_manager.rb:180:in `process_args'
	from /usr/lib/ruby/vendor_ruby/rubygems/command_manager.rb:149:in `run'
	from /usr/lib/ruby/vendor_ruby/rubygems/gem_runner.rb:53:in `run'
	from /bin/gem:13:in `<main>'
```

thanks!

**Comment by lefte** [comment]
I'm too green to Ruby to understand how to repair this, is it an issue with the methadone version? It no longer installs with any version of Ruby 2 or 3 for me.

**Comment by matthutchinson** [comment]
Hmm, apologies both. I've dropped my contributions to maintaining this gem in recent years.  I'll try to pick this up soon and look at refreshing some of the dependencies, like methadone, which is at this point, very old.

**Comment by matthutchinson** [comment]
I've just tried myself (latest macOS and latest Ruby 3.3.1) - it installs and runs fine.

**Comment by lefte** [comment]
LOL you might have fixed it with the 0.16.5 release today? Before I was trying it on 0.16.4:

```shell
ruby -v
ruby 3.3.1 (2024-04-23 revision c56cd86388) [arm64-darwin23]
```

```shell
uname -a
Darwin heimdall.lefte.com 23.4.0 Darwin Kernel Version 23.4.0: Fri Mar 15 00:10:42 PDT 2024; root:xnu-10063.101.17~1/RELEASE_ARM64_T6000 arm64
```

```shell
rbenv global 3.3.1
```

```shell
gem install lolcommits
Fetching lolcommits-0.16.5.gem
Fetching git-1.19.1.gem
Successfully installed git-1.19.1
Successfully installed lolcommits-0.16.5
Parsing documentation for git-1.19.1
Installing ri documentation for git-1.19.1
Parsing documentation for lolcommits-0.16.5
Installing ri documentation for lolcommits-0.16.5
Done installing documentation for git, lolcommits after 0 seconds
2 gems installed

A new release of RubyGems is available: 3.5.9 → 3.5.11!
Run `gem update --system 3.5.11` to update your installation.
```
